code/datums/components/forensics.dm

```diff
@@ -0,0 +1,129 @@
+/datum/component/forensics
+	var/list/fingerprints		//assoc print = print
+	var/list/hiddenprints		//assoc ckey = realname/gloves/ckey
+	var/list/blood_DNA			//assoc dna = bloodtype
+	var/list/fibers				//assoc print = print
+
+/datum/component/forensics/proc/return_fingerprints()
+	return length(fingerprints)? fingerprints.Copy() : list()
+
+/datum/component/forensics/proc/return_hiddenprints()
+	return length(hiddenprints)? hiddenprints.Copy() : list()
+
+/datum/component/forensics/proc/return_blood_DNA()
+	return length(blood_DNA)? blood_DNA.Copy() : list()
+
+/datum/component/forensics/proc/return_fibers()
+	return length(fibers)? fibers.Copy() : list()
+
+/datum/component/forensics/proc/wipe_fingerprints()
+	fingerprints = null
+
+/datum/component/forensics/proc/wipe_hiddenprints()
+	return	//no.
+
+/datum/component/forensics/proc/wipe_blood_DNA()
+	blood_DNA = null
+
+/datum/component/forensics/proc/wipe_fibers()
+	fibers = null
+
+/datum/component/forensics/proc/add_fingerprints(list/_fingerprints)	//list(text)

```

Comment:
Suggestion change this proc name to add_known_fingerprints since this one is all done with copying prints already from a object.

Reply:
intended